File: cpp/n268_MissNum.cpp
// https://leetcode-cn.com/problems/missing-number/
#include <iostream>
#include <vector>
#include <string>
using namespace std;

class Solution {
public:
    int missingNumber(vector<int>& nums) {
        size_t l = nums.size();
        int sum = l * (l + 1)/2;
        for (auto s:nums) sum -= s;
        return sum;
    }
};

int main(){
    vector<int> nums = {0, 5, 3, 4, 1};
    Solution solu;
    cout << solu.missingNumber(nums) << endl;
}

File: python/n268_MissNum.py
# https://leetcode-cn.com/problems/missing-number/
class Solution:
    def missingNumber(self, nums) -> int:
        l = len(nums)
        sum = l * (l+1) / 2
        for i in nums:
            sum -= i
        return int(sum)


if __name__ == "__main__":
    nums = [0, 3, 5, 2, 1]
    solu = Solution()
    print(solu.missingNumber(nums))